src/app/state/accountHistorySlice.ts

```diff
@@ -212,4 +265,95 @@ export const selectOrderHistory = createSelector(
   (orderHistory) => orderHistory.filter((order) => order.status !== "PENDING")
 );
 
+// A function that checks an order against a filter condition and returns TRUE if it matches, FALSE otherwise
+type FilterFunction = (order: adex.OrderReceipt) => boolean;
+
+let selectCombinedOrders = (filterFunction: FilterFunction) =>
+  createSelector(
+    (state: RootState) => state.accountHistory.orderHistory,
+    (state: RootState) => state.accountHistory.orderHistoryAllPairs,
+    (orderHistory, orderHistoryAllPairs) => {
+      // Create a Map to handle duplicates
+      const orderMap = new Map<number, adex.OrderReceipt>();
+
+      // Add orders from orderHistory
+      orderHistory.forEach((order) => {
+        orderMap.set(order.id, order);
+      });
+
+      // Add orders from orderHistoryAllPairs, will overwrite any duplicates from orderHistory
+      orderHistoryAllPairs.forEach((order) => {
+        orderMap.set(order.id, order);
+      });
```

Comment:
Since the order Id's are not unique among pairs but only within a given pair (e.g. every pair starts the ordercount at 1, so every pair has a single order with the id 1), we need to add the name or address to the key of the hashmap, to make the orders truly unique. For example:

```ts
      // Generate unique orderId helper
      const getUniqueOrderId = (order: Order) => `${order.name}_${order.id}`;

      // Add orders from orderHistory
      orderHistory.forEach((order) => {
        orderMap.set(getUniqueOrderId(order), order);
      });

      // Add orders from orderHistoryAllPairs, will overwrite any duplicates from orderHistory
      orderHistoryAllPairs.forEach((order) => {
        orderMap.set(getUniqueOrderId(order), order);
      });
```